packages/cozy-harvest-lib/src/components/TriggerManager.jsx

```diff
@@ -66,29 +66,35 @@ export class TriggerManager extends Component {
       account
     })
 
-    let folder
-    if (konnectors.needsFolder(konnector)) {
-      const path = `${t('default.baseDir')}/${konnector.name}/${slugify(
-        accounts.getLabel(account)
-      )}`
-
-      folder =
-        (await statDirectoryByPath(path)) || (await createDirectoryByPath(path))
-
-      await addPermission(konnector, konnectors.buildFolderPermission(folder))
-      await addReferencesTo(konnector, [folder])
+    try {
+      let folder
+
+      if (konnectors.needsFolder(konnector)) {
+        const path = `${t('default.baseDir')}/${konnector.name}/${slugify(
+          accounts.getLabel(account)
+        )}`
+
+        folder =
+          (await statDirectoryByPath(path)) ||
+          (await createDirectoryByPath(path))
+
+        await addPermission(konnector, konnectors.buildFolderPermission(folder))
+        await addReferencesTo(konnector, [folder])
+      }
```

Comment:
Could be extracted outside the component in a  ensureKonnectorHasFolderIfNeeded(konnector) method